src/SFA.DAS.EmployerIncentives.Queries/NewApprenticeIncentive/GetApprenticeshipEligibility/GetApprenticeshipEligibilityQueryHandler.cs

```diff
@@ -1,25 +1,29 @@
 ﻿using System.Threading;
 using System.Threading.Tasks;
 using SFA.DAS.EmployerIncentives.Abstractions.Queries;
+using SFA.DAS.EmployerIncentives.Data;
 using SFA.DAS.EmployerIncentives.Domain.Services;
 using SFA.DAS.EmployerIncentives.Queries.Extensions;
 
 namespace SFA.DAS.EmployerIncentives.Queries.NewApprenticeIncentive.GetApprenticeshipEligibility
 {
     public class GetApprenticeshipEligibilityQueryHandler : IQueryHandler<GetApprenticeshipEligibilityRequest, GetApprenticeshipEligibilityResponse>
     {
-        private INewApprenticeIncentiveEligibilityService _eligibilityService;
+        private readonly INewApprenticeIncentiveEligibilityService _eligibilityService;
+        private readonly IUlnQueryRepository _queryRepository;
 
-        public GetApprenticeshipEligibilityQueryHandler(INewApprenticeIncentiveEligibilityService eligibilityService)
+        public GetApprenticeshipEligibilityQueryHandler(INewApprenticeIncentiveEligibilityService eligibilityService, IUlnQueryRepository queryRepository)
         {
             _eligibilityService = eligibilityService;
+            _queryRepository = queryRepository;
         }
 
         public async Task<GetApprenticeshipEligibilityResponse> Handle(GetApprenticeshipEligibilityRequest query, CancellationToken cancellationToken = default)
         {
             var apprenticeship = query.Apprenticeship.ToApprenticeship();
+            var ulnUsed = await _queryRepository.UlnAlreadyOnSubmittedIncentiveApplication(query.Apprenticeship.UniqueLearnerNumber);
 
-            var isEligible = _eligibilityService.IsApprenticeshipEligible(apprenticeship);
+            var isEligible = !ulnUsed && _eligibilityService.IsApprenticeshipEligible(apprenticeship);
```

Comment:
I'm wondering if it'd be worth moving all of the logic to the eligibility service.  That'd mean having a repo method to get any existing claimapprenticeship records for the uln and passing them to the service and allowing it to decide what to do with them.

Reply:
That's an interesting point. I wanted to do that originally, but the UlnQueryRepo class needs the Db Context which exists in Data project. We can do it by putting the interface into the abstractions layer and let the IoC resolve the class. 

Happy to do it this way if you want to. We could call the interface/class IUlnValidator

Reply:
done